troposphere/asg-elb-dns.py

```diff
@@ -110,81 +94,121 @@ def generate_cloudformation_template():
         Default="300",
     ))
 
-    health_check_target = template.add_parameter(Parameter(
-        "LoadBalancerHealthCheckTarget",
-        Type="String",
-    ))
-
-    health_check_interval = template.add_parameter(Parameter(
-        "LoadBalancerHealthCheckInterval",
-        Type="String",
-    ))
-
-    health_check_timeout = template.add_parameter(Parameter(
-        "LoadBalancerHealthCheckTimeout",
-        Type="String",
-    ))
-
-    healthy_threshold = template.add_parameter(Parameter(
-        "LoadBalancerHealthyThreshold",
-        Type="String",
-    ))
-
-    unhealthy_threshold = template.add_parameter(Parameter(
-        "LoadBalancerUnHealthyThreshold",
-        Type="String",
-    ))
-
-    enable_connection_draining = template.add_parameter(Parameter(
-        "LoadBalancerEnableConnectionDraining",
-        Type="String",
-        Default="True",
-    ))
-
-    connection_draining_timeout = template.add_parameter(Parameter(
-        "LoadBalancerConnectionDrainingTimeout",
-        Type="String",
-        Default="30",
-    ))
-
-    launchconfigurationname = template.add_parameter(Parameter(
-        "LaunchConfigurationName",
-        Type="String",
-    ))
-
-    new_weight = template.add_parameter(Parameter(
-        "NewDnsWeight",
-        Type="String",
-        Default="100",
-    ))
-
-    loadbalancer = template.add_resource(elb.LoadBalancer(
-        "LoadBalancer",
-        ConnectionDrainingPolicy=elb.ConnectionDrainingPolicy(
-            Enabled=Ref(enable_connection_draining),
-            Timeout=Ref(connection_draining_timeout),
-        ),
-        Subnets=Ref(subnet),
-        HealthCheck=elb.HealthCheck(
-            Target=Ref(health_check_target),
-            HealthyThreshold=Ref(healthy_threshold),
-            UnhealthyThreshold=Ref(unhealthy_threshold),
-            Interval=Ref(health_check_interval),
-            Timeout=Ref(health_check_timeout),
-        ),
-        Listeners=[
-            elb.Listener(
-                LoadBalancerPort="80",
-                InstancePort="80",
-                Protocol="HTTP",
-                InstanceProtocol="HTTP",
+    if enable_elb:
```

Comment:
Github ate my previous comment here. Let's try again:

I'm not sure what kind of "improvements" do you want to do here. The first thing that I came up with is to encapsulate the ELB code and have a _merge_ function, for two given templates. Example:

```python

def elb():
    template = Template()
    # ...
    return template

def merge(template_1, template_2):
    # bla bla
    return template

# your code
template = merge(template, elb()) if enable_elb else template
```

If you want to extend this practice to "anything" else, you could create a dictionary with the available options and then apply the functions that you want, based on the user's input. Example:


```python
def asg():
    pass
def elb():
    pass

ELB = 'elb'
ASG = 'asg'

AVAILABLE_FUNCS = {ELB: elb, ASG: asg}


# Argument parsing
to_apply = []
if enabled_elb:
    to_apply.append(ELB)

# then you could apply them dinamically
for func in to_apply:
    template = merge(template, func())
```

It's possible to make the argument parsing even more dynamic. 

One other option is to have use a _strategy pattern_ in a _dict_ that applies a function or another depending on its value. Example:

```python

def elb():
    pass

ELB_STRATEGY = {
    True: elb,
    False: Template()
}

# Argument parsing

elb_strategy = ELB_STRATEGY[bool(enabled_elb)]

# bla bla bla

template = merge(template, elb_strategy)
template = merge(template, asg_strategy)
#...
```

I'm not sure if this is what you were looking for. If it's not, we can talk about it tomorrow :)